deploy/azure/infra/main.tf

```diff
@@ -80,35 +80,122 @@ resource "azurerm_data_factory_managed_private_endpoint" "blob_pe" {
   subresource_name   = "blob"
 }
 
+resource "null_resource" "approve_adf_blob_private_endpoint" {
```

Comment:
I think "azapi_resource" is a better approach https://github.com/hashicorp/terraform-provider-azurerm/issues/19777 because:
- its native terraform
- you don't have to authenticate with azure again using "az login.."

Reply:
Hi Bal, I was looking into this in a separate branch [](https://github.com/Ensono/stacks-azure-data/tree/bug/adf-pe-approvalv1) and managed to get it working for storage accounts, ADLS,ADLS and databricks, but facing some challenges with SQL and KV seems like some API issue that will require some time on this. I think we can take this into further enhancement if we want to implement this.